app/services/scenes/scenes.ts

```diff
@@ -266,6 +266,16 @@ class ScenesViews extends ViewHandler<IScenesState> {
 
     return false;
   }
+
+  createAndAddSource(
+    sceneId: string,
+    name: string,
+    sourceType: TSourceType,
+    settings: Dictionary<any>,
+  ) {
+    const scene = this.getScene(sceneId);
+    return scene?.createAndAddSource(name, sourceType, settings).sceneItemId;
+  }
```

Comment:
we really shouldn't be altering state in a View model, that's very much an antipattern, but is there a specific reason you did it here?

Reply:
This is now using the Scenes Service, is still an indirection to the view since we seem to need the `#createAndAddSource` from the scene object returned by that view's `getScene`. Is only for convenience though.